### PR TITLE
Feature/delete candidate model

### LINKED ISSUE #6: Create Candidate model
Create entity and its attributes.

### LINKED ISSUE #23: Candidate CRUD operations
Define repository and expose EP + CRUD

### PR DESCRIPTION
**Pull request by polgodia**
CRUD Delete Candidate

### CODE REVIEW

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/DeleteCandidateStepDefs.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    public DeleteCandidateStepDefs(StepDefs stepDefs, CandidateRepository candidateRepository) {
+        this.stepDefs = stepDefs;
+        this.candidateRepository = candidateRepository;
+    }
+
+
+    @When("I delete a candidate with name {string}")
+    public void iDeleteCandidateTo(String name) throws Throwable {
+
+        Candidate candidate = new Candidate();
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/DeleteCandidateStepDefs.java:33]: It is better to create the candidate in a separate @Given step like `There exist a candidate with name...` and then here recover the candidate using a findByName and then its URI using the .getUri() method

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/DeleteCandidateStepDefs.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        stepDefs.result = stepDefs.mockMvc.perform(
+                delete(newResourceUri)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(stepDefs.mapper.writeValueAsString(candidate))
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate())
+        ).andDo(print());
+    }
+
+    @And("The previously deleted candidate doesn't exist")
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/DeleteCandidateStepDefs.java:48]: This step recovers the recently created candidate but doesn't check if it exists or not. Better to directly check using the findByName method of the repository and then assert that the returned list is empty. The step would be something like @Then `There is no candidate with name...`

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public UpdateCandidateStepDefs(StepDefs stepDefs, CandidateRepository candidateRepository) {
+        this.stepDefs = stepDefs;
+        this.candidateRepository = candidateRepository;
+    }
+
+
+    @When("I change the name of the candidate to {string}")
+    public void iChangeNameOfCandidateTo(String name) throws Throwable {
+
+        Candidate candidate = new Candidate();
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java]: Like in the case of Delete (in fact you can reuse the step and its definition) use a separate step to state that there is an existing candidate with given name. Then, this step should just update the name for that candidate. You can recover it using its current name provided in the step, like @When `I change the name of the candidate with name ... to ...`

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                patch(newResourceUri)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(stepDefs.mapper.writeValueAsString(candidate))
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate())
+        ).andDo(print());
+    }
+
+    @And("The previously updated candidate has now name {string}")
+    public void thePreviouslyUpdatedCandidateHasNowName(String name) throws Throwable {
```
Comment (rogargon) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java]: You can check, reusing existing step, that there is no candidate with the old name and that there is a candidate with the new one. In both cases, you can use a findByName, no need to go through the API in the @Given and @Then steps

File: src/test/resources/features/DeleteCandidate.feature [hunk 5]
```diff
@@ -0,0 +1,15 @@
+Feature: Candidate Delte
```
Comment (rogargon) [inline src/test/resources/features/DeleteCandidate.feature]: Candidate Delete

**Review by rogargon — CHANGES_REQUESTED** [review]

**Review by bergacat1 — APPROVED** [review]

**Pull request merged**